### ISSUE TITLE
Add normalization for DataLoader

### ISSUE THREAD

**Issue by EmGarr** [NathanRxl/concept_drift#4]
There is no  normalization which is mandatory for few learning algorithms (except Decision trees).

**Comment by NathanRxl** [comment]
I strongly think the scaler must be independent from both the DataLoader and the predicting algorithms. Is it possible to use StandardScaler from scikit-learn in this way instead ? :

pseudocode:

    for batch in StreamGenerator():
        normalized_batch = StandardScaler().fit_transform(batch)
        algorithm1.update(batch)              # This algorithm *do not* need the scaling
        algorithm2.update(normalized_batch)   # This algorithm *do* need the scaling
        algorithm3.update(normalized_batch)   # This algorithm *do* need the scaling
        algorithm4.update(batch)              # This algorithm *do not* need the scaling

**Comment by EmGarr** [comment]
For me we cannot just do normalization like that. We need to keep track of the previous samples at least through a sliding window.

**Comment by NathanRxl** [comment]
*The algorithm* (perhaps) needs to keep track of the previous samples at least through a sliding window. This "memory" must appear in the algorithm, not at this level. The algorithm is then free to store the normalized_batch as it needs it.

**Comment by EmGarr** [comment]
How can you perform a Standard deviation normalization if you just consider one batch. To me it doesn't make sense and it  is impossible  for one example. This memory is not related to the algorithm. 
This publication provides few methods for feature scaling https://arxiv.org/abs/1305.6646
We won't have the time to implement them but it is worth to know it.

**Comment by NathanRxl** [comment]
Your algorithm do not know the future rest of the stream at time t, you only know what is coming and what came. You can do a StandardScaling at time t on a subset of batches t, t-1, ..., 1 if you want but keep being aware the concept can drift (that is precisely our case). It does not make sense to do a Scaling of the whole stream before serving it if it is what you have in mind.

**Comment by NathanRxl** [comment]
@EmGarr This issue has been solved by 66007e4ab6fd9bf948ae5d0bd30cfb02fadebbd9, is not it ?
